Add Galaxy Imaging Hackathon 2025 blog post (#3129)
galaxyproject/galaxy-hub: content/news/12-05-2025-GalaxyImagingHackathon2025/index.md

---
title: "Lights, Camera... Hacking! Highlights from the Galaxy Imaging Hackathon 2025"
authors: "Diana Chiang Jurado" 
contributions:

>     funding: [NFDI4BioImage,de.KCD]

Reviewer: can we add OSCARS-FIESTA here?

Author: Yes, we can! Please do so

Author: I added it already @beatrizserrano, thanks lots!

tags: [imaging, hackathon, community, workflows, FAIR, microscopy]
layout: news
date: "2025-05-12"
tease: "Three days of collaborative hacking, coffee-fueled brainstorming, and community-powered innovation—Galaxy Imaging is growing, and so are its FAIR tools!" 
organizers:
  - galaxyfreiburg
  - eurobioimaging
supporters:
  - nfdi
  - oscars
subsites: [all,global,esg,eu]
---

# 🧬 Lights, Camera... Hacking! Highlights from the Galaxy Imaging Hackathon 2025 🎬🔬

> ## April 22nd-25th, 2025 | Freiburg, Germany 🌱

Reviewer 2: ```suggestion
## April 22nd-24th, 2025 | Freiburg, Germany 🌱
```

Reviewer 2: I think it was 22-24th, no?

Author: Yes, you are right! Thanks @nilchia <3!

> In April 2025, members of the Galaxy community came together for an energizing **three-day hackathon** dedicated to advancing FAIR image analysis and workflows in Galaxy. Whether virtually or in-person—and accompanied by delicious coffee and sweets—we rolled up our sleeves, shared ideas, solved bugs, built tools, and wrapped up each day with great conversations over dinner in Freiburg. 🍽️

Reviewer 3: ```suggestion
In April 2025, members of the Galaxy community came together for an energizing **three-day hackathon** dedicated to advancing FAIR image analysis and workflows in Galaxy. Whether virtually or in-person — and accompanied by delicious coffee and sweets — we rolled up our sleeves, shared ideas, solved bugs, built tools, and wrapped up each day with great conversations over dinner in Freiburg. 🍽️
```

> This wasn’t just about closing issues—it was about shared problem solving and community-driven development. Across those three days, contributors dove into discussions, whiteboarded bold ideas, and tackled technical challenges side by side. By the end, we had made meaningful progress toward expanding Galaxy’s capabilities for image analysis!

Reviewer 3: ```suggestion
This wasn’t just about closing issues — it was about shared problem solving and community-driven development. Across those three days, contributors dove into discussions, whiteboarded bold ideas, and tackled technical challenges side by side. By the end, we had made meaningful progress toward expanding Galaxy’s capabilities for image analysis!
```

## Key Milestones

> A new [Galaxy workflow for Brightfield Microscopy](https://github.com/FAIR-imaging/galaxy-image-community/issues/35), tailored to Masson Trichrome and IHC images, was successfully completed. The [Bellavista visualiser](https://github.com/FAIR-imaging/galaxy-image-community/issues/45) was integrated into Galaxy, bringing spatial transcriptomics data to life. We also released a new [tutorial on parameter tuning and image segmentation evaluation](https://github.com/FAIR-imaging/galaxy-image-community/issues/38), helping users benchmark and optimise segmentation results.

Reviewer 3: ```suggestion
A new [Galaxy workflow for Brightfield Microscopy](https://github.com/FAIR-imaging/galaxy-image-community/issues/35), tailored to Masson Trichrome and IHC images, was successfully completed. The [Bellavista visualiser](https://github.com/FAIR-imaging/galaxy-image-community/issues/45) was integrated into Galaxy, bringing spatial transcriptomics data to life. We also released a new [tutorial on parameter tuning and image segmentation evaluation](https://github.com/FAIR-imaging/galaxy-image-community/issues/38), helping users to benchmark and optimise segmentation results.
```

Author: The "to" is not needed here =)

> A new [Galaxy workflow for Brightfield Microscopy](https://github.com/FAIR-imaging/galaxy-image-community/issues/35), tailored to Masson Trichrome and IHC images, was successfully completed. The [Bellavista visualiser](https://github.com/FAIR-imaging/galaxy-image-community/issues/45) was integrated into Galaxy, bringing spatial transcriptomics data to life. We also released a new [tutorial on parameter tuning and image segmentation evaluation](https://github.com/FAIR-imaging/galaxy-image-community/issues/38), helping users benchmark and optimise segmentation results.

Reviewer 3: There was AE above (e.g., "energizing"), so lets stick to AE:
```suggestion
A new [Galaxy workflow for Brightfield Microscopy](https://github.com/FAIR-imaging/galaxy-image-community/issues/35), tailored to Masson Trichrome and IHC images, was successfully completed. The [Bellavista visualiser](https://github.com/FAIR-imaging/galaxy-image-community/issues/45) was integrated into Galaxy, bringing spatial transcriptomics data to life. We also released a new [tutorial on parameter tuning and image segmentation evaluation](https://github.com/FAIR-imaging/galaxy-image-community/issues/38), helping users benchmark and optimize segmentation results.
```

> Meanwhile, our tools were tested across diverse file types in a [format compatibility investigation](https://github.com/FAIR-imaging/galaxy-image-community/issues/34), and improvements were made to enhance the user experience—such as restoring the [GitHub link on imaging.usegalaxy.eu](https://github.com/FAIR-imaging/galaxy-image-community/issues/29) and [standardising `.TIFF` vs `.TIF` extensions](https://github.com/FAIR-imaging/galaxy-image-community/issues/43) across tools.

Reviewer 3: …AE again:
```suggestion
Meanwhile, our tools were tested across diverse file types in a [format compatibility investigation](https://github.com/FAIR-imaging/galaxy-image-community/issues/34), and improvements were made to enhance the user experience — such as restoring the [GitHub link on imaging.usegalaxy.eu](https://github.com/FAIR-imaging/galaxy-image-community/issues/29) and [standardizing `.TIFF` vs `.TIF` extensions](https://github.com/FAIR-imaging/galaxy-image-community/issues/43) across tools.
```

> We also took the time to highlight training materials—most notably with the publication of a feature article to promote the BMZ microscopy tutorial, which you can [read here](https://galaxyproject.org/news/2025-04-23-bmz-galaxy/). In parallel, we began designing a general-purpose [“Where to Start” tutorial for image analysis](https://github.com/FAIR-imaging/galaxy-image-community/issues/36) to support newcomers in navigating Galaxy’s imaging landscape.

Reviewer 3: Correct me if I'm mistaken here, but I think the formatting of `"` as something more fancy like `“` is up to the Markdown renderer, and shouldn't be done in Markdown for better machine readability:
```suggestion
We also took the time to highlight training materials — most notably with the publication of a feature article to promote the BMZ microscopy tutorial, which you can [read here](https://galaxyproject.org/news/2025-04-23-bmz-galaxy/). In parallel, we began designing a general-purpose ["Where to Start" tutorial for image analysis](https://github.com/FAIR-imaging/galaxy-image-community/issues/36) to support newcomers in navigating Galaxy’s imaging landscape.
```

> We also took the time to highlight training materials — most notably with the publication of a feature article to promote the BMZ microscopy tutorial, which you can [read here](https://galaxyproject.org/news/2025-04-23-bmz-galaxy/). In parallel, we began designing a general-purpose ["Where to Start" tutorial for image analysis](https://github.com/FAIR-imaging/galaxy-image-community/issues/36) to support newcomers in navigating Galaxy’s imaging landscape.

Reviewer 4: HI @dianichj thanks a lot for this!

One tutorial was actually finalized and already it is already online 
https://training.galaxyproject.org/training-material/topics/imaging/tutorials/parameter-tuning/tutorial.html
I think it worth it to cite it

Author: Thanks so much @rmassei for noticing, I can't believe I missed it. I added it to the post. Tutorial fantastico!

## Exciting Work Underway

> Development began on integrating [OMERO as a file source](https://github.com/FAIR-imaging/galaxy-image-community/issues/17), as well as [porting Fractal tasks and workflows](https://github.com/FAIR-imaging/galaxy-image-community/issues/31) into the Galaxy ecosystem. We explored extending support for AI inference tools with [TensorFlow compatibility](https://github.com/FAIR-imaging/galaxy-image-community/issues/37), started building a [VPT preprocessing tool for MEROSCOPE images](https://github.com/FAIR-imaging/galaxy-image-community/issues/41), and developed plans for a new tool focused on [gel band quantification using ImageJ](https://github.com/FAIR-imaging/galaxy-image-community/issues/42). A proposal to implement a [more robust TIFF viewer](https://github.com/FAIR-imaging/galaxy-image-community/issues/44) is also underway to better support high-resolution image inspection.

Reviewer 3: I'd leave this out because it seems that this was a dead end, unfortunately:
```suggestion
Development began on integrating [OMERO as a file source](https://github.com/FAIR-imaging/galaxy-image-community/issues/17), as well as [porting Fractal tasks and workflows](https://github.com/FAIR-imaging/galaxy-image-community/issues/31) into the Galaxy ecosystem. We explored extending support for AI inference tools with [TensorFlow compatibility](https://github.com/FAIR-imaging/galaxy-image-community/issues/37), started building a [VPT preprocessing tool for MEROSCOPE images](https://github.com/FAIR-imaging/galaxy-image-community/issues/41), and developed plans for a new tool focused on [gel band quantification using ImageJ](https://github.com/FAIR-imaging/galaxy-image-community/issues/42).
```

## Final Thoughts

> Throughout it all, tool integration, workflow refinement, and in-depth investigation defined the rhythm of the hackathon. We tested pipelines, cleaned up quirks, and kept usability and accessibility front and centre.

Reviewer 3: …and AE again:
```suggestion
Throughout it all, tool integration, workflow refinement, and in-depth investigation defined the rhythm of the hackathon. We tested pipelines, cleaned up quirks, and kept usability and accessibility front and center.
```

> What excites us most isn’t just the code we wrote—it’s the momentum of a growing, inclusive community. The Galaxy imaging community is expanding, and together we're building a welcoming, collaborative space to support image analysis in a FAIR, scalable, and user-friendly way.

Reviewer 3: …about the fancy vs. standard formatting, I assume this should be handled by the renderer (forgive me if I'm mistaken on this)
```suggestion
What excites us most isn't just the code we wrote — it's the momentum of a growing, inclusive community. The Galaxy imaging community is expanding, and together we're building a welcoming, collaborative space to support image analysis in a FAIR, scalable, and user-friendly way.
```
